server/src/termdb.scatter.js

```diff
@@ -440,10 +442,15 @@ export async function getScatterCoordinates(req, q, ds) {
 
 	sql += `SELECT ax.sample AS sampleId, ax.value as x, ay.value AS y , ${zterm ? 'az.value AS z' : '0 as z'}
 			FROM anno_${xterm.type} ax 
-			JOIN anno_${yterm.type} ay on ax.sample = ay.sample  
+			JOIN anno_${yterm.type} ay on ax.sample = ay.sample  JOIN sampleidmap on ax.sample = sampleidmap.id
 			${zterm ? `JOIN anno_${zterm.type} az on ax.sample = az.sample and az.term_id = '${zterm.id}' ` : ''}
 			WHERE ax.term_id = '${xterm.id}' and ay.term_id = '${yterm.id}'`
 	if (filter) sql += ` AND ax.sample IN ${filter.CTEname}`
+
+	const cols = listTableColumns(ds.cohort.db.connection, 'sampleidmap')
```

Comment:
the table columns should be queried once during server init, and stored as part of ds.cohort.db, maybe like
```
ds.cohort.db.tables = {
  sampleidmap: {cols: [...]},
  sample_ancestry: {...}
}
```